packages/react-menu/src/components/Menu/useMenu.tsx

```diff
@@ -123,3 +106,57 @@ const useMenuSelectableState = (state: MenuState) => {
     });
   });
 };
+
+const useMenuOpenState = (state: MenuState) => {
+  const shouldHandleKeyboadRef = React.useRef(false);
+  const onOpenChange: MenuState['onOpenChange'] = useEventCallback((e, data) => state.onOpenChange?.(e, data));
+
+  const [open, setOpen] = useControllableValue(state.open, state.defaultOpen);
+  // TODO fix useControllableValue typing
+  state.open = open !== undefined ? open : state.open;
```

Comment:
(nit): this does not look like a typing issue, can you please update/remove TODO?

Reply:
you're right, the hook should rather support initial value